src/apis/avm/utxos.ts

```diff
@@ -757,86 +981,121 @@ export class UTXOSet {
-    // Fees are burned, so no toAddresses, only feeSenderAddresses and changeAddresses, both are the feeSenderAddresses
-    const utx:UnsignedTx = this.buildBaseTx(
-      networkid, blockchainid, fee, [], feeSenderAddresses, feeSenderAddresses, feeAssetID, undefined, asOf
-    );
-    const ins:Array<TransferableInput> = utx.getTransaction().getIns();
-    const outs:Array<TransferableOutput> = utx.getTransaction().getOuts();
-    const exportOuts:Array<TransferableOutput> = [];
-    for (let i:number = 0; i < utxoids.length; i++) {
-      const utxo:UTXO = this.getUTXO(utxoids[i]);
-      const assetID:Buffer = utxo.getAssetID();
-      const output:AmountOutput = utxo.getOutput() as AmountOutput;
-      const amt:BN = output.getAmount().clone();
-      const txid:Buffer = utxo.getTxID();
-      const outputidx:Buffer = utxo.getOutputIdx();
-      const input:SecpInput = new SecpInput(amt);
-      const xferin:TransferableInput = new TransferableInput(txid, outputidx, assetID, input);
-      const fromAddresses:Array<Buffer> = output.getAddresses(); // Verify correct approach
-      const spenders:Array<Buffer> = output.getSpenders(fromAddresses, asOf);
-      for (let j = 0; j < spenders.length; j++) {
-        const idx:number = output.getAddressIdx(spenders[j]);
-        if (idx === -1) {
-          /* istanbul ignore next */
-          throw new Error('Error - UTXOSet.buildBaseTx: no such '
-          + `address in output: ${spenders[j]}`);
-        }
-        xferin.getInput().addSignatureIdx(idx, spenders[j]);
-      }
-      ins.push(xferin);
-      const xferOut:TransferableOutput = new TransferableOutput(assetID, output);
-      exportOuts.push(xferOut)
+    let ins:Array<TransferableInput> = [];
+    let outs:Array<TransferableOutput> = [];
+    let exportouts:Array<TransferableOutput> = [];
+    
+    if(typeof changeAddresses === "undefined") {
+      changeAddresses = toAddresses;
+    }
+
+    const zero:BN = new BN(0);
+    
+    if (amount.eq(zero)) {
+      return undefined;
+    }
+
+    if(typeof feeAssetID === "undefined") {
+      feeAssetID = avaxAssetID;
+    } else if (feeAssetID.toString("hex") !== avaxAssetID.toString("hex")) {
+      /* istanbul ignore next */
+      throw new Error('Error - UTXOSet.buildExportTx: '
+      + `feeAssetID must match avaxAssetID`);
+    }
+
+    const aad:AssetAmountDestination = new AssetAmountDestination(toAddresses, fromAddresses, changeAddresses);
+    if(avaxAssetID.toString("hex") === feeAssetID.toString("hex")){
+      aad.addAssetAmount(avaxAssetID, amount, fee);
+    } else {
+      aad.addAssetAmount(avaxAssetID, amount, zero);
+      aad.addAssetAmount(feeAssetID, zero, fee);
     }
-    const exportTx:ExportTx = new ExportTx(networkid, blockchainid, outs, ins, memo, exportOuts);
+    const success:Error = this.getMinimumSpendable(aad, asOf, locktime, threshold);
+    if(typeof success === "undefined") {
+      ins = aad.getInputs();
+      exportouts = aad.getOutputs();
```

Comment:
This will send the change outputs to the peer chain in addition to the amounts that were requested to be sent.

Reply:
Oof. That's a boner. I'll fix it.